web/modules/custom/employees/src/Plugin/QueueWorker/UserSyncWorker.php

```diff
@@ -0,0 +1,139 @@
+use Drupal\user\Entity\User;
+use Drupal\portland_openid_connect\Util\PortlandOpenIdConnectUtil;
+use Exception;
+
+/**
+ * SyncUsersWorker class.
+ *
+ * A worker plugin to consume items from "user_sync"
+ * and synchronize users from Entra ID.
+ *
+ * @QueueWorker(
+ *   id = "user_sync",
+ *   title = @Translation("Synchronize Users Queue"),
+ *   cron = {"time" = 120}
+ * )
+ */
+class UserSyncWorker extends QueueWorkerBase implements ContainerFactoryPluginInterface
+{
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition)
+  {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition)
+  {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @param array $data
+   * [
+   *   "access_token" => $access_token,
+   *   "domain" => $domain,
+   *   "users" => users[{...}]
+   * ]
+   * 
+   * User data example:
+   * {
+   *     "id": "1234dfc6-310c-4d5e-aa62-6237a8222f1b",
+   *     "accountEnabled": true,
+   *     "userPrincipalName": "John.Doe@portlandoregon.gov",
+   *     "displayName": "Doe, John",
+   *     "givenName": "John",
+   *     "surname": "Doe",
+   *     "jobTitle": "City worker",
+   *     "mail": "John.Doe@portlandoregon.gov",
+   *     "companyName": "Bureau of Technology Services",
+   *     "department": "DepartmentA",
+   *     "officeLocation": "123/1234",
+   *     "businessPhones": [
+   *       *  "555-555-5555"
+   *     ],
+   *     "mobilePhone": "555-555-5555",
+   *     "employeeId": "1234567",
+   *     "userType": "Member",
+   *     "employeeType": "Regular",
+   *     "streetAddress": "1120 SW 5th Ave, Suite 1234",
+   *     "city": "Portland",
+   *     "state": "OR",
+   *     "postalCode": "97204",
+   *     "mailNickname": "jdoe"
+   * }
+   */
+  public function processItem($data)
+  {
+    foreach ($data["users"] as $user_data) {
+      // Skip accounts without first name, last name, userPrincipalName, or email. These are not people acount.
+      if (
+        empty($user_data['givenName']) ||
+        empty($user_data['surname']) ||
+        empty($user_data['mail']) ||
+        empty($user_data['userPrincipalName']) ||
+        empty($user_data['id']) ||
+        str_ends_with($user_data['userPrincipalName'], 'onmicrosoft.com') ||
+        str_contains(strtolower($user_data['mail']), '_adm@')
+      ) {
+        continue;
+      }
+
+      // Look up user by Drupal user name (Principal name in AD)
+      // Sometimes a user will be recreated with the same principal name but different AD ID
+      $users = \Drupal::entityTypeManager()->getStorage('user')->loadByProperties(['name' => $user_data['userPrincipalName']]);
```

Comment:
Shouldn't this use the trimmed principal name in case the name was trimmed when the user was created (line 109)?

Reply:
Added the trimming call. Also updated the trim function to use MD5 hash instead of inserting a 3 digit random number to make sure the result is consistent.